sdk/core/azure-core-amqp/CMakeLists.txt

```diff
@@ -160,25 +152,111 @@ set(AZURE_CORE_AMQP_SOURCE
+#  src/impl/uamqp/models/private/performatives
+#  src/impl/uamqp/models/private/properties_impl.hpp
+#  src/impl/uamqp/models/private/source_impl.hpp
+#  src/impl/uamqp/models/private/target_impl.hpp
+#  src/impl/uamqp/models/private/value_impl.hpp
+#  src/impl/uamqp/models/private/performatives/detach_impl.hpp
+#  src/impl/uamqp/models/private/performatives/transfer_impl.hpp
+  src/impl/uamqp/network/amqp_header_transport.cpp
+  src/impl/uamqp/network/sasl_transport.cpp
+  src/impl/uamqp/network/socket_listener.cpp
+  src/impl/uamqp/network/socket_transport.cpp
+  src/impl/uamqp/network/tls_transport.cpp
+  src/impl/uamqp/network/transport.cpp
+  src/impl/uamqp/network/private/transport_impl.hpp
+)
+endif()
+
+if(USE_RUST_AMQP)
+set(AZURE_RUST_AMQP_SOURCE
+  src/impl/rust_amqp/amqp/claim_based_security.cpp
+  src/impl/rust_amqp/amqp/connection.cpp
+  src/impl/rust_amqp/amqp/connection_string_credential.cpp
+  src/impl/rust_amqp/amqp/link.cpp
+  src/impl/rust_amqp/amqp/management.cpp
+  src/impl/rust_amqp/amqp/message_receiver.cpp
+  src/impl/rust_amqp/amqp/message_sender.cpp
+  src/impl/rust_amqp/amqp/session.cpp
+  src/impl/rust_amqp/amqp/private/claims_based_security_impl.hpp
+  src/impl/rust_amqp/amqp/private/connection_impl.hpp
+  src/impl/rust_amqp/amqp/private/link_impl.hpp
+  src/impl/rust_amqp/amqp/private/management_impl.hpp
+  src/impl/rust_amqp/amqp/private/message_receiver_impl.hpp
+  src/impl/rust_amqp/amqp/private/message_sender_impl.hpp
+  src/impl/rust_amqp/amqp/private/session_impl.hpp
+#  src/impl/rust_amqp/models/amqp_detach.cpp
+#  src/impl/rust_amqp/models/amqp_error.cpp
+#  src/impl/rust_amqp/models/amqp_header.cpp
+#  src/impl/rust_amqp/models/amqp_message.cpp
+#  src/impl/rust_amqp/models/amqp_properties.cpp
+#  src/impl/rust_amqp/models/amqp_transfer.cpp
+#  src/impl/rust_amqp/models/amqp_value.cpp
+#  src/impl/rust_amqp/models/message_source.cpp
+#  src/impl/rust_amqp/models/message_target.cpp
+#  src/impl/rust_amqp/models/messaging_values.cpp
+#  src/impl/rust_amqp/models/private/error_impl.hpp
+#  src/impl/rust_amqp/models/private/header_impl.hpp
+#  src/impl/rust_amqp/models/private/message_impl.hpp
+#  src/impl/rust_amqp/models/private/performatives
+#  src/impl/rust_amqp/models/private/properties_impl.hpp
+#  src/impl/rust_amqp/models/private/source_impl.hpp
+#  src/impl/rust_amqp/models/private/target_impl.hpp
+#  src/impl/rust_amqp/models/private/value_impl.hpp
+#  src/impl/rust_amqp/models/private/performatives/detach_impl.hpp
+#  src/impl/rust_amqp/models/private/performatives/transfer_impl.hpp
+)
+
+endif()
+
+if (USE_UAMQP)
+add_library(azure-core-amqp ${AZURE_CORE_AMQP_SOURCE} ${AZURE_CORE_AMQP_HEADER} ${AZURE_UAMQP_SOURCE} $<TARGET_OBJECTS:uamqp>)
 elseif(USE_RUST_AMQP)
-add_library(azure-core-amqp ${AZURE_CORE_AMQP_SOURCE} ${AZURE_CORE_AMQP_HEADER})
+add_library(azure-core-amqp ${AZURE_CORE_AMQP_SOURCE} ${AZURE_CORE_AMQP_HEADER} ${AZURE_RUST_AMQP_SOURCE})
```

Comment:
Are `USE_UAMQP` and `USE_RUST_AMQP` mutually exclusive? If so, should the above `set(AZURE_RUST_AMQP_SOURCE ...` also be in an if/elseif rather than two independent if blocks? It might be easier to read and be consistent if we combine all the if/elseif into one block.

Reply:
Yes - One says to use the Rust AMQP implementation, the other says to use the UAMQP implementation. 

And you can't have an ELSIF inside a set statement - they have to be separate set statements. The way this is currently structured is that there is a set of common elements, a set of UAMQP elements and a set of Rust AMQP elements.